internal/keygen/elgamal.go

```diff
@@ -27,34 +27,39 @@ type ElGamal struct {
 	Y *big.Int // public key
 	G *big.Int // generator
 	P *big.Int // modulus
-	Q *big.Int // order
+	Q *big.Int // (P - 1) / 2
 }
 
 // adapted from https://github.com/dlitz/pycrypto/blob/master/lib/Crypto/PublicKey/ElGamal.py
 func NewElGamal(modulusLength int) (*ElGamal, error) {
 	p, err := emmy.GetSafePrime(modulusLength)
-	g := big.NewInt(0)
+	// q = (p - 1) / 2
+	q := new(big.Int).Div(new(big.Int).Sub(p, big.NewInt(1)), big.NewInt(2))
 
 	if err != nil {
 		return nil, fmt.Errorf("failed to generate safe prime")
 	}
 
+	g := big.NewInt(0)
```

Comment:
Initialize this by doing `g := new(big.Int)` - I think an explicit zero initialization makes it less clear what's going on.

Reply:
Done.